getfastq.sh

```diff
@@ -102,41 +103,63 @@ function _progress_getfastq {
         printf "Bad TARGETS path '$1'.\n"
         exit 1 # Argument failure exit status: bad target path
     fi
-
-    local log_file="$(find "${target_dir}" \
-        -maxdepth 1 -type f -iname "Z_getFASTQ_*.log")"
-
-    if [[ -n "$log_file" ]]; then
-
-        # NOTE: the -- is used here to indicate the end of 'bash' command
-        #       options and the beginning of Bash script arguments.
-        # NOTE: possible non-matching greps inside 'find ... -exec bash' won't
-        #       break the script, even when 'set -e' is active.
-        printf "\n${grn}Completed:${end}\n"
-        find "${target_dir}" -maxdepth 1 -type f -iname "Z_getFASTQ_*.log" \
-            -exec bash -c '
-                grep -E " saved \[| already there;" "$1"
-            ' -- {} \;
-
-        printf "\n${red}Failed:${end}\n"
-        find "${target_dir}" -maxdepth 1 -type f -iname "Z_getFASTQ_*.log" \
-            -exec bash -c '
-                grep -E ".+Terminated| unable to |Not Found." "$1"
-            ' -- {} \;
-
-        printf "\n${yel}Incoming:${end}\n"
-        find "${target_dir}" -maxdepth 1 -type f -iname "Z_getFASTQ_*.log" \
-            -exec bash -c '
-                dead_track=$(tail -n 3 "$1" | grep -E \
-                    " saved \[| already there;|Terminated|unable to|Not Found")
-                [[ -z $dead_track ]] && tail -n 1 "$1" && echo
-            ' -- {} \;
-        exit 0 # Success exit status
-    else
+    
+    declare -a logs=()
+    readarray -t logs < <(find "${target_dir}" -maxdepth 1 -type f -iname "Z_getFASTQ_*.log")
+    if [[ "${#logs[@]}" -eq 0 ]]; then
         printf "No getFASTQ log file found in '${target_dir}'.\n"
         exit 2 # Argument failure exit status: missing log
     fi
-}
+
+    declare -a completed=()
+    declare -a failed=()
+    declare -a incoming=()
+    for log in "${logs[@]}"; do
+        # From most important to least important
+        test_failed=$(c1grep -E ".+Terminated| unable to |Not Found.|Unable to" "$log")
+        if [[ -n $test_failed ]]; then
+            failed+=("$test_failed")
+            continue
+        fi
+        test_completed=$(c1grep -E " saved \[| already there;" "$log" | tail -n 1 )
+        if [[ -n $test_completed ]]; then
+            completed+=("$test_completed")
+            continue
+        fi
+        
+        incoming+=( "$(tail -n1 "$log" | rev | cut -d$'\r' -f 1 | rev )" )
+    done
+
+    printf "\n${grn}Completed:${end}\n"
+    if [ ${#completed[@]} -eq 0 ]; then
+        printf "\t- No completed items!\n"
+    else
+        for item in "${completed[@]}"; do
+            printf "\t- ${item}\n"
+        done
+    fi
+
+    printf "\n${red}Failed:${end}\n"
+    if [ ${#failed[@]} -eq 0 ]; then
+        printf "\t- No failed items!\n"
+    else
+        for item in "${failed[@]}"; do
+            printf "\t- ${item}\n"
```

Comment:
I had trouble with the formatting for special characters. Using `%b`causes some characters that we want to *not* escape to be escaped, so I think we should be careful. `echo` might be the best option, since we want to send the messages to `stdout` anyway.

Reply:
yeah, lets go with `echo`!